src/redux/userSessions/userSessions.js

```diff
@@ -0,0 +1,135 @@
+const LOGIN_SUCCESS = 'book-vehicle/userSessions/LOGIN_SUCCESS';
+const LOGIN_FAILURE = 'book-vehicle/userSessions/LOGIN_FAILURE';
+const LOGOUT_REQUEST = 'book-vehicle/userSessions/LOGOUT_REQUEST';
+const LOGOUT_SUCCESS = 'book-vehicle/userSessions/LOGOUT_SUCCESS';
+const LOGOUT_ERROR = 'book-vehicle/userSessions/LOGOUT_ERROR';
+const SET_USER = 'book-vehicle/userSessions/SET_USER';
+
+const initialUser = {
+  id: null,
+  username: null,
+  email: null,
+  role: null,
+};
+
+const defaultState = {
+  loading: false,
+  error: null,
+  user: initialUser,
+};
+
+export default function reducer(state = defaultState, action = {}) {
+  switch (action.type) {
+    case LOGIN_REQUEST:
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    case LOGIN_SUCCESS:
+      return {
+        ...state,
+        loading: false,
+        error: null,
+        user: action.payload,
+      };
+    case LOGIN_FAILURE:
+      return {
+        ...state,
+        loading: false,
+        error: action.payload,
+      };
+    case LOGOUT_REQUEST:
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    case LOGOUT_SUCCESS:
+      return {
+        ...state,
+        loading: false,
+        error: null,
+        user: initialUser,
+      };
+    case LOGOUT_ERROR:
+      return {
+        ...state,
+        loading: false,
+        error: action.payload,
+      };
+    case SET_USER:
+      return {
+        ...state,
+        user: action.payload,
+      };
+    default:
+      return state;
+  }
+}
+
+const loginRequest = () => ({ type: LOGIN_REQUEST });
+const loginSuccess = (userData) => ({ type: LOGIN_SUCCESS, payload: userData });
+const loginFailure = (error) => ({ type: LOGIN_FAILURE, payload: error });
+const logoutRequest = () => ({ type: LOGOUT_REQUEST });
+const logoutSuccess = () => ({ type: LOGOUT_SUCCESS });
+const logoutError = (error) => ({ type: LOGOUT_ERROR, payload: error });
+
+export const setUser = (user) => ({ type: SET_USER, payload: user });
+
+export const login = (user, navigate) => async (dispatch) => {
+  dispatch(loginRequest());
+  try {
+    const data = JSON.stringify(user);
+    const response = await client.post('users/sign_in', data);
+    const resData = response.data;
+    const token = resData.data.authentication_token;
+    TokenManager.setToken(token);
+    UserObjectManager.setUserObject(data);
+
+    dispatch(loginSuccess(data));
+    navigate('/main');
+    toast.success("You've successfully logged in!");
+  } catch (error) {
+    dispatch(loginFailure(error.response.data.message));
+    toast.error(
+      `There was an error logging in: ${error.response.data.message}`,
+    );
+  }
+};
+
+export const logout = (navigate) => async (dispatch) => {
+  dispatch(logoutRequest());
+  try {
+    //  const response = await client.delete('/users/sign_out');
+    //  console.log('SIGN OUT RESONSE SEVER WITH AXIOS:', response);
```

Comment:
- [ ] Kindly remove this old piece of commented code, in order to improve the cleanliness of your repo  💪